src/views/tutorial-view/utils/variants/context.tsx

```diff
@@ -0,0 +1,52 @@
+import {
+	createContext,
+	useState,
+	useContext,
+	useMemo,
+	ReactNode,
+	Dispatch,
+	SetStateAction,
+	useEffect,
+} from 'react'
+import { TutorialVariant } from './types'
+
+type VariantContextValue = {
+	activeVariant?: TutorialVariant
+	setActiveVariant?: Dispatch<SetStateAction<TutorialVariant>>
+}
+
+interface VariantProviderProps {
+	children: ReactNode
+	variant?: TutorialVariant
+}
+
+export function useVariants(): VariantContextValue {
+	return useContext(VariantContext)
+}
+
+const VariantContext = createContext({ activeVariant: null })
+VariantContext.displayName = 'VariantContext'
+
+export default function VariantProvider({
+	children,
+	variant,
+}: VariantProviderProps) {
+	const [activeVariant, setActiveVariant] = useState<TutorialVariant>(variant)
```

Comment:
💭 might just rename this to `variant` & `setVariant`, since `activeOption` is a prop and that feels repetitive.

Reply:
👆 I dig that!